### DISCUSSION TITLE
Windows Terminal Preview v1.14.143

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#13157] [Announcements]
Welcome to the terminaldome! Today this release page is graced by the works of pinch hitter (and pinch release notes writer) @carlos-zamora.

# Features
* Windows Terminal now has better support for the xterm "Alternate Screen Buffer", and can now handle alternate scroll mode and resize/reflow better (#12561) (#12569) (#12719)
* Using windowed applications from the terminal should work much better now
	* ConPTY can now handle show/hide window calls and even knows who the window owner is (#12515) (#12526)
	* Windows created by console apps now appear above the terminal (#12799) (#12899)
	* Focus events are now sent through VT input (#12900)
* You can now use the `experimental.useBackgroundImageForWindow` (bool, default `false`) global setting to apply one background image for your entire window! (#12893) (#13114) (thanks @nico-abram!)
* You are now able to select all the text in the buffer using the `selectAll` action. This is bound by default to <kbd>ctrl+shift+a</kbd> (#13045) (#13084)

# Changes
## UI
* @dansmor7 contributed some visual changes to the tabs, scrollbar, new tab button, caption buttons, color picker, settings UI, command palette, and search box to move us closer to the Windows 11 design language. Really just about any WinUI surface we have, it's been polished up! (thanks @dansmor7!) (#12913) (#12916) (#12973) (#13083) 
* `RadioButton`s in the settings UI have been replaced with `ComboBox`es. This gives an added bonus to keyboard and screen reader users, and makes it easier to navigate through and change these settings. (#12833)  

## Interactivity
* The IME input mode now defaults to English when interacting with Windows Terminal  (#13028) (thanks @YanceyChiew!)
* Terminal is now aware of toggled state for  <kbd>Caps Lock</kbd>, <kbd>Scroll Lock</kbd>, and <kbd>Num Lock</kbd>  (#12823) (thanks @matkaas!) 

## Settings
* There's now a VERY EXPERIMENTAL new VT passthrough mode setting that makes ConPTY do minimal translations and may make your terminal a little faster 🏃💨 and a lot more broken! (#11264) (#13051) (#13109)
	* Use the  `experimental.connection.passthroughMode` (bool, default `true`) profile setting and it should be set on the profile's next launch
	* ⚠️ WARNING ⚠️ This seems to mostly work with CMD and WSL. PowerShell is mostly sad 😭.
* The `trimBlockSelection` global setting now defaults to `true` (#12737) 
* Terminal now ignores `newTab` actions with a profile index greater than the number of profiles (#11621) 

## Atlas Renderer Improvements
* ClearType is no longer always enabled (#12705)
* The grayscale blending shader should now be working properly (#12734)
* OpenConsole's leak check report should be fixed now (#12415)
* The shader power draw was reduced using explicit branching (#12552)
* The renderer is now smarter about when to resize the buffer when scrolling (#13100)

## Documentation
* Our GitHub repo now supports rich code navigation (#12855) (#12910) (#12910). Bug reports are also automatically tagged as bugs (#12404) (thanks @snxx-lppxx!)
* We've uploaded specs for Theme-controlled color scheme switching (#12613) (thanks @arkthur!) and Default Terminal (#7414)
* The README has been updated to mention the required .NET Targeting Pack (#12896) (thanks @pizzaz93!)
* Words are hard! Thanks to @sebastiansterk, @DimitriPapadopoulos, and @jsoref for making sure we use the right words and grammar across our repo. (#12386) (#12475) (#12835)
* `bellSound` is now in the schema (#13035) (thanks @pizzaz93!) 

# Bug Fixes
* Terminal should be able to find Cascadia Mono... Third time's the charm? 🍀 (#12904) 
* `commandline` in `profile.defaults` should no longer override the `commandline`s of profiles that specify cmd.exe or powershell.exe. (#12906) 
* Get rid of a memory leak in onecore interactivity (#12340)  
* We should be maintaining the virtual viewport bottom properly now (#12972) (#13052) (#13087) (thanks @j4james!)
* Screen readers can now read some settings in the UI better (#13032) 
* Replace "acrylic" with "acrylic material" for localization purposes (#12505) 
* The "close tab" button color now matches the tab text color (#13018) (thanks @ianjoneill!) 

## Reliability
* Fix a crash when deleting the last profile in the settings UI (#13044) 
* Fix resize crash in OpenConsole when using the Atlas Renderer (#13015)

## Code health and Maintainability
* @j4james unified the terminal and console's VT handlers, deleting thousands of lines of redundant code in the process! (#12207) (#12247) (#12389) (#12390) (#12568) (#12703) (#13024) (#13039) (thanks @j4james!)
* Added some missing breaks to cases in `IslandWindow` (#12926) (thanks @jmelas!)
* Added the MIT license for a few files that were missing it (#12368) (thanks @jerry-shao!)
* Replaced `sizeof` with `ARRAYSIZE` in `SystemConfigurationProvider` (#12273) (thanks @abdoulkkonate!)
* Use type inference throughout the project (#12975)
* Use memcmp for TextAttribute & TextColor comparison (#10566) (thanks @skyline75489!)
* Thanks to @ianjoneill, @achermack, @YanceyChiew, @EmJayGee, @dmachaj, and @j4james for performing a ton of build system, code health and maintainability improvements!

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.14.1432.0'>Windows Terminal Preview v1.14.143</a>.</em>

**Comment by etoyz** [comment]
LOVE IT😍

**Reply by sonvoba** [reply]
Add-AppxPackage : Deployment failed with HRESULT: 0x80073CF3
![Add-AppxPackage  Deployment failed with HRESULT 0x80073CF3](https://user-images.githubusercontent.com/10577963/170792964-98196abb-2e3a-489c-a53e-903888909e7f.png)

**Reply by DHowett** [reply]
You are missing a package dependency which is documented in the release notes. :)

**Reply by sonvoba** [reply]
URL/Link to the "release notes" please (is there such a thing?)

**Reply by DHowett** [reply]
o_O This discussion _is_ and also contains another link to the release notes..?

~~https://github.com/microsoft/terminal/releases/tag/v1.14.1432.0~~

~~Check out the "Why are there so many packages?" detail disclosure.~~

**Reply by DHowett** [reply]
Honestly, your best bet given that the VC++ team will not update that redistributable is to download the "Preinstallation Kit" from [here](https://github.com/microsoft/terminal/releases/download/v1.13.11431.0/Microsoft.WindowsTerminal_Win11_1.13.11432.0_8wekyb3d8bbwe.msixbundle_Windows10_PreinstallKit.zip) and install the `vclibs` package inside it that matches your architecture. Sorry that this isn't more clear.

**Comment by abdallahmansour6** [comment]
Thank you

**Comment by trueuto** [comment]
Approx three days after that release I get my app updated via Microsoft Store but its version is reported as 1.14.1451.0. It is not the first time that the app after the update has different (higher) number than release notes mention. Do you have notes for this version too? Is it now a standard process that the version available via Store is different than released here?

**Reply by zadjii-msft** [reply]
Sorry, we've been a little distracted and haven't had the chance to write the release notes for 1.14.145. 1.14.145 is different than 1.14.143, it includes a couple hotfixes to 143. Namely #13164, but there was something else IIRC.

**Reply by trueuto** [reply]
Thanks for your reply! I tried to get some info about that from commits/tags here in GitHub, but it seems that releases are in fact being prepared **outside** of GitHub as latest tag is 1.14.143. I have found that some information is visible under the "release-1.14" branch in the commit log (by dates), though.

**Reply by DHowett** [reply]
Ugh, I'm sorry I never uploaded 1.14.145 to the GitHub releases page. It's built off commit 997b4acce5b6de63f9414340ccdfa9b9e7bc2ad7.

Releases aren't exactly prepared inside or outside GitHub in our case; the pipeline that we use to stamp out production builds runs in Azure DevOps, pulls directly from GitHub, and communicates build state back to GitHub... but tagging has historically been entirely manual, because we produce a whole bunch of intermediate builds that we _don't_ want to clutter up the tag history with.

It's been almost entirely on me to (1) do the release engineering and (2) document the release engineering so that it is not almost entirely on me to (1) do the release engineering. :smile:

**Reply by DHowett** [reply]
_Boom!_ https://github.com/microsoft/terminal/releases/tag/v1.14.1451.0

**Reply by trueuto** [reply]
Thanks!
